training_agent_demo.py

```diff
@@ -61,12 +63,27 @@ def print_weights(network, title):
 
     all_score.append(score)
     average_score = np.mean(all_score[-100:])
+    average_score_1k = np.mean(all_score[-1000:])
+    
+    if i % 1000 == 0 and i > 0:  # Check every 1000 episodes
+        print(f"Eposide {i} | Average Last 1k Rollout Score: {average_score_1k} | Current Level: {gym_env.current_level} ")
+        if average_score >= PERFORMANCE_THRESHOLD:  # Increase level if performance is good
+            gym_env.increase_difficulty()
```

Comment:
we want to be able to use this function with stable_baseline3 learn() function, and it wont be calling this increse / decrease difficulty intelligently, so can we migrate these 2 functions to the new_gym()? maybe put this inside the reset()? such that if it reaches goal or add something like cumulated score, then the gym can auto adjust its difficulty within itself.